packages/gravity/blade-runner/src/plan/env/scheduler-env.ts

```diff
@@ -0,0 +1,216 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Callback, Redis, type RedisOptions } from 'ioredis';
+import path from 'node:path';
+
+import { Trigger } from '@dxos/async';
+import { log } from '@dxos/log';
+
+import { ReplicantRpcHandle, open, close } from './replicant-rpc-handle';
+import { REDIS_PORT, createRedisRpcPort } from './util';
+import { WebSocketRedisProxy } from './websocket-redis-proxy';
+import { type ReplicantBrain, type SchedulerEnv, type RpcHandle, type ReplicantClass } from '../interface';
+import { type ProcessHandle, runBrowser, runNode } from '../run-process';
+import {
+  type ReplicantRuntimeParams,
+  type ReplicantParams,
+  type GlobalOptions,
+  type TestParams,
+  AGENT_LOG_FILE,
+  type ReplicantsSummary,
+} from '../spec';
+
+export class SchedulerEnvImpl<S> implements SchedulerEnv {
```

Comment:
Duplicated code between this and ReplicatorEnvImpl - unify via composition/inheritance

Reply:
Added TODO for it